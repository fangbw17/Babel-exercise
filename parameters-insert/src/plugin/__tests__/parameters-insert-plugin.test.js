const pluginTester = require('babel-plugin-tester')
const insertPlugin = require('../parameters-insert-plugin')


pluginTester({
    plugin: insertPlugin,
    babelOptions: {
        parserOpts: {
            sourceType: "unambiguous",
            plugins: ["jsx"],
        },
    },
    tests: {
        "console.xx前插入了CallExpression的AST": {
            code: `
                console.log(1);

                function func() {
                    console.log(2);
                }

                export default class Clazz {
                    say() {
                        console.debug(3);
                    }
                    render() {
                        return <div>{console.error(4)}</div>
                    }
                }
            `,
            snapshot: true,
        },
    },
});
