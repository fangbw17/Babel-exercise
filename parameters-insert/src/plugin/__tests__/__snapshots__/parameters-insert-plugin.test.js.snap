// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unknown plugin 1. console.xx前插入了CallExpression的AST: 1. console.xx前插入了CallExpression的AST 1`] = `


console.log(1);

function func() {
    console.log(2);
}

export default class Clazz {
    say() {
        console.debug(3);
    }
    render() {
        return <div>{console.error(4)}</div>
    }
}
            

      ↓ ↓ ↓ ↓ ↓ ↓

console.log(
  "/Users/aaron/Desktop/Lpz/code/Babel/parameters-insert/src/plugin/__tests__/parameters-insert-plugin.test.js: (2, 0)"
);
console.log(1);
function func() {
  console.log(
    "/Users/aaron/Desktop/Lpz/code/Babel/parameters-insert/src/plugin/__tests__/parameters-insert-plugin.test.js: (5, 4)"
  );
  console.log(2);
}
export default class Clazz {
  say() {
    console.log(
      "/Users/aaron/Desktop/Lpz/code/Babel/parameters-insert/src/plugin/__tests__/parameters-insert-plugin.test.js: (10, 8)"
    );
    console.debug(3);
  }
  render() {
    return (
      <div>
        {[
          console.log(
            "/Users/aaron/Desktop/Lpz/code/Babel/parameters-insert/src/plugin/__tests__/parameters-insert-plugin.test.js: (13, 21)"
          ),
          console.error(4),
        ]}
      </div>
    );
  }
}

`;
